### DISCUSSION TITLE
UI Mockups/Concept for header bar

### DISCUSSION THREAD

**Discussion by VikSin** [akiraux/Akira#475] [Ideas]
### Mode Concept for changing options in headerbar

![Frame](https://user-images.githubusercontent.com/25245765/65072180-fb49d400-d9ad-11e9-8f98-d0590efb51ae.png)
#### Concept to create different modes
As per elementary os guidelines their is no menubar. Showing all options right in front of UI
this can create problems in upcoming designs where to add buttons and option.
Modes can be helpfull to hide and show options by how you handling the object in which situation.  

![Frame 2](https://user-images.githubusercontent.com/25245765/65072672-018c8000-d9af-11e9-94a4-c831fda1f112.png)
#### Let’s look into each mode
- **Object**
Inserting artboard, shapes ,vector, text , pencil and images.
Scaling , rotating, moving obect. options like changing border radius with nob
Aligning, changing styles 

![Frame (1)](https://user-images.githubusercontent.com/25245765/65073149-e1a98c00-d9af-11e9-86ff-87bf3962b301.png)

- **Edit**
Editing Shapes / More options to deform shapes 
options like bending node nob into line, corner, Smooth, Symmetric and random seed for node
Even options to merge to node points, delete, rip, and add node points in curve

![Frame (3)](https://user-images.githubusercontent.com/25245765/65073776-23870200-d9b1-11e9-932d-4b9ba06d012b.png)

- **Draw** 
Vector Painting mode vector brushes to draw and paint 

![Frame (4)](https://user-images.githubusercontent.com/25245765/65073996-955f4b80-d9b1-11e9-88c2-8dcecc218bff.png)

> Doubble click on object it will enter in edit mode and dubble click outside object it will exit out of edit mode. For switching to other modes we have to select from dropdown in headerbar.

And Thanks to all Developers For bringing this tool to linux.

**Comment by Alecaddd** [comment]
Hey, thank you so much for your interest in Akira and for this insightful proposal.

Different modes to change the headerbar and adapt the UI based on the current "status" of an item is something I've been thinking for a while.
I'm not completely sold on this approach as I'd like to try to offer this type of flexibility in the left and right panels, showing and hiding toolboxes based on the current status of an item.

> When the user double clicks on a shape, he enters the node mode and a node panel appears in the left sidebar that shows all the various available options for that status.

I'm not entirely sold on the headerbar approach because some modes won't need that much space, and if all those options can be better organized into dynamic side panels, we can avoid changing the headerbar UI, which might be a bit disorienting for first time users.

Where I see this approach shining, is the Slicing/Export area, which I think is where Affinity Designer shines compared to Sketch.

Anyway, this is a great conversation I'd like to keep going for future versions of Akira. For many technical and timing reasons, unfortunately, we won't be able to implement this approach in the first release.
